### PR TITLE
Add new `Style/KeywordArgumentsMerging` cop

### LINKED ISSUE #13252: New cop to detect unnecessary `merge` when passing keyword arguments
```ruby
# bad
some_method(foo, **options.merge(bar: 1))

# good
some_method(foo, **options, bar: 1)
```

I found (grep by `\*\*\w+\.merge` regexp) 12 such offences in the `rails/rails` codebase and 14 in my own gem (https://github.com/fatkodima/online_migrations).

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #13252.

```ruby
# bad
some_method(foo, **options.merge(bar: 1))
some_method(foo, **options.merge(other_options).merge(more_options))
some_method(foo, **options.merge({ bar: 1 }))
some_method(foo, **options.merge(bar: 1, baz: 2))
some_method(foo, **options.merge(bar: 1, baz: 2, **other_hash))
some_method(foo, **options.merge(other_options))

# good
some_method(foo, **options, bar: 1)
some_method(foo, **options, **other_options, **more_options)
some_method(foo, **options, bar: 1)
some_method(foo, **options, bar: 1, baz: 2)
some_method(foo, **options, bar: 1, baz: 2, **other_hash)
some_method(foo, **options, **other_options)
```

### CODE REVIEW

File: lib/rubocop/cop/style/keyword_arguments_merging.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+module RuboCop
+  module Cop
+    module Style
+      # When passing an existing hash as keyword arguments, add additional arguments
+      # directly rather than using `merge`.
+      #
+      # @example
+      #   # bad
+      #   some_method(**opts.merge(foo: true))
+      #   some_method(**opts.merge(other_opts))
```
[the remark this answers is not in the document]
Comment (fatkodima) [reply lib/rubocop/cop/style/keyword_arguments_merging.rb:12]: Added 👍

File: lib/rubocop.rb [hunk 2]
```diff
@@ -559,6 +559,7 @@
 require_relative 'rubocop/cop/style/inline_comment'
 require_relative 'rubocop/cop/style/invertible_unless_condition'
 require_relative 'rubocop/cop/style/ip_addresses'
+require_relative 'rubocop/cop/style/keyword_arguments_merging'
```
[the remark this answers is not in the document]
Comment (fatkodima) [reply lib/rubocop.rb:562]: But we actually deal with arguments here?

File: lib/rubocop/cop/style/keyword_arguments_merging.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # When passing an existing hash as keyword arguments, add additional arguments
+      # directly rather than using `merge`.
```
Comment (bbatsov) [inline lib/rubocop/cop/style/keyword_arguments_merging.rb:7]: You might want to add some explanation here as to why using `merge` is a bad idea. And a matching style guide entry.

File: lib/rubocop/cop/style/keyword_arguments_merging.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      #   some_method(**opts.merge(other_opts))
+      #
+      #   # good
+      #   some_method(**opts, foo: true)
+      #   some_method(**opts, **other_opts)
+      #
+      class KeywordArgumentsMerging < Base
+        extend AutoCorrector
+
+        MSG = 'Add additional arguments directly rather than using `merge`.'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/keyword_arguments_merging.rb:21]: I'd probably say "List/Provide" instead of "Add" in this context.

Comment (fatkodima) [reply lib/rubocop/cop/style/keyword_arguments_merging.rb]: Changed.

File: lib/rubocop/cop/style/keyword_arguments_merging.rb (hunk 3 above)
Comment (fatkodima) [reply lib/rubocop/cop/style/keyword_arguments_merging.rb:7]: Added explanation.

> And a matching style guide entry

But it already has an example from a style guide?

Comment (bbatsov) [reply lib/rubocop/cop/style/keyword_arguments_merging.rb:7]: 🤦 My bad!

**Pull request merged**